Tracks Properties/graysonsolis_Toggle track pitch envelope.lua

```diff
@@ -0,0 +1,144 @@
+end
+
+----------------------------------------------------------------------------------------
+-- REAPITCH FX SETUP
+----------------------------------------------------------------------------------------
+-- TrackFX_GetCount before adding to detect new addition
+local before = reaper.TrackFX_GetCount(track)
+
+-- Silently add/find ReaPitch (Cockos)
+local fx_idx = reaper.TrackFX_AddByName(track, "ReaPitch (Cockos)", false, 0)
+if fx_idx < 0 then
+  fx_idx = reaper.TrackFX_AddByName(track, "ReaPitch (Cockos)", false, 1)
+end
+if fx_idx < 0 then
+  reaper.ShowMessageBox("Could not add/find ReaPitch.", "Error", 0)
+  reaper.PreventUIRefresh(0)
+  return
+end
+
+-- Always hide its FX chain (silent)
+reaper.TrackFX_Show(track, fx_idx, 2)
+
+-- Determine if the FX was just added
+local added = (reaper.TrackFX_GetCount(track) > before)
+--[[
+----------------------------------------------------------------------------------------
+-- ENVELOPE ACCESS
+----------------------------------------------------------------------------------------
+
+CUSTOMIZATION TIP
+
+To allow user selection of a different ReaPitch parameter envelope, change the value
+of the `param` variable below to one of the following:
+
+    0:  Wet
+    1:  Dry
+    2:  1: Enabled
+    3:  1: Shift (full range)
+    4:  1: Shift (cents)         <-- current default
+    5:  1: Shift (semitones)
+    6:  1: Shift (oct)
+    7:  1: Formant adjust (full range)
+    8:  1: Formant adjust (cents)
+    9:  1: Formant adjust (semitones)
+    10: 1: Volume
+    11: 1: Pan
+    12: Bypass
+    13: Wet
+    14: Delta
+
+For example, to toggle the envelope for "Shift (semitones)", set:
+
+    local param = 5
+]]
+
+---
+local param = 4  -- Shift (cents) YOU CAN CHANGE THIS!!!!
+local env = reaper.GetFXEnvelope(track, fx_idx, param, true)
+if not env then
+  reaper.ShowMessageBox("Failed to access envelope.", "Error", 0)
+  reaper.PreventUIRefresh(0)
+  return
+end
+
+-- Read the envelope chunk once
+local ok, chunk = reaper.GetEnvelopeStateChunk(env, "", false)
+if not ok then
+  reaper.ShowMessageBox("Failed to read envelope.", "Error", 0)
+  reaper.PreventUIRefresh(0)
+  return
+end
+
+----------------------------------------------------------------------------------------
+-- TOGGLE LOGIC
+----------------------------------------------------------------------------------------
+local show
+if added then
+  -- Newly added FX → always show envelope
+  show = true
+else
+  -- Existing FX → toggle based on current VIS flags
+  local v1,v2,v3 = chunk:match("VIS (%d+) (%d+) (%d+)")
+  show = not (v1 == "1" and v2 == "1" and v3 == "1")
+end
+
+----------------------------------------------------------------------------------------
+-- UPDATE CHUNK FLAGS (ONE-PASS)
+----------------------------------------------------------------------------------------
+chunk = chunk
+  :gsub("ACT %d+",         "ACT " .. (show and 1 or 0))
+  :gsub("VIS %d+ %d+ %d+", "VIS " .. (show and "1 1 1" or "0 0 0"))
```

Comment:
This force-disables the "Show envelope in lane" setting. Is this intended?

Also the third value should always be 1 (see [p=2198426](https://forum.cockos.com/showpost.php?p=2198426)).